### PR TITLE
Fixes when neither LEDA nor GMP is available

### PR DESCRIPTION
**Pull request by sloriot**

### CODE REVIEW

File: Cone_spanners_2/test/Cone_spanners_2/CMakeLists.txt [hunk 1]
```diff
@@ -5,12 +5,16 @@ cmake_minimum_required(VERSION 3.1...3.23)
...
-  RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}
-  ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp)
-foreach(cppfile ${cppfiles})
-  create_single_source_cgal_program("${cppfile}")
-endforeach()
+create_single_source_cgal_program("cones_inexact.cpp")
+create_single_source_cgal_program("theta_inexact.cpp")
+create_single_source_cgal_program("yao_inexact.cpp")
+
+if(CGAL_Core_FOUND OR LEDA_FOUND)
```
Comment (lrineau) [inline Cone_spanners_2/test/Cone_spanners_2/CMakeLists.txt:14]: That is strange. `find_package(CGAL REQUIRED COMPONENTS Core)` should stop the compilation if `Core` is not found. That is certainly an error in `CGALConfig.cmake`.

File: Arrangement_on_surface_2/test/Arrangement_on_surface_2/cgal_test.cmake [hunk 2]
```diff
@@ -72,13 +72,24 @@ if(CGAL_DISABLE_GMP)
...
-  message(STATUS "GMP is disable. Try to use LEDA instead.")
-  set(GMPZ_NT ${LEDA_INT_NT})
-  set(QUOTIENT_CGAL_GMPZ_NT ${LEDA_RAT_NT})
-  set(CGAL_GMPQ_NT ${LEDA_RAT_NT})
-  set(LAZY_CGAL_GMPQ_NT ${LAZY_LEDA_RAT_NT})
-  set(LAZY_GMPZ_NT ${LAZY_LEDA_RAT_NT})
-  set(CGAL_GMPZ_NT ${LEDA_INT_NT})
+  message(STATUS "GMP is disable.")
+  if (CGAL_USE_LEDA)
+    message(STATUS "Try to use LEDA instead.")
```
Comment (albert-github) [inline Arrangement_on_surface_2/test/Arrangement_on_surface_2/cgal_test.cmake:77]: `"Try" -> `"  Try` as message is an explanation of the previous message.
(also acounts for the `MP case).

File: Arrangement_on_surface_2/test/Arrangement_on_surface_2/cgal_test_with_cmake [hunk 3]
```diff
@@ -78,13 +78,24 @@ CORE_INT_NT=15
...
+      CGAL_GMPZ_NT=$LEDA_INT_NT
+    else
+      echo Try to use MP float instead.
+      GMPZ_NT=$MP_FLOAT_NT
+      QUOTIENT_CGAL_GMPZ_NT=$QUOTIENT_MP_FLOAT_NT
+      CGAL_GMPQ_NT=$QUOTIENT_MP_FLOAT_NT
+      LAZY_CGAL_GMPQ_NT=$LAZY_QUOTIENT_MP_FLOAT_NT
+      LAZY_GMPZ_NT=$LAZY_QUOTIENT_MP_FLOAT_NT
+      CGAL_GMPZ_NT=$MP_FLOAT_NT
+    fi
```
Comment (albert-github) [inline Arrangement_on_surface_2/test/Arrangement_on_surface_2/cgal_test_with_cmake:98]: see remarks with cgal_test.cmake

File: Straight_skeleton_2/test/Straight_skeleton_2/CMakeLists.txt [hunk 4]
```diff
@@ -5,23 +5,34 @@ cmake_minimum_required(VERSION 3.1...3.23)
...
+  create_single_source_cgal_program("test_sls_weighted_polygons_with_holes.cpp")
+  create_single_source_cgal_program("issue7149.cpp")
+  if(CGAL_Qt6_FOUND)
+    target_link_libraries(issue7149 PUBLIC CGAL::CGAL_Basic_viewer)
+    target_link_libraries(issue7284 PUBLIC CGAL::CGAL_Basic_viewer)
+    target_link_libraries(test_sls_offset PUBLIC CGAL::CGAL_Basic_viewer)
+    target_link_libraries(test_sls_weighted_polygons_with_holes PUBLIC CGAL::CGAL_Basic_viewer)
+  endif()
+else()
+  message("NOTICE: Some test require CGAL_Core (or LEDA), and will not be compiled.")
```
Comment (albert-github) [inline Straight_skeleton_2/test/Straight_skeleton_2/CMakeLists.txt:37]: Isn't there a more flexible way possible?
(see e.g. https://github.com/doxygen/doxygen/blob/master/examples/CMakeLists.txt and look for BASIC_EXAMPLES. still a list but a bit more flexible, maybe something is possible here as well...)

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.0-Ic-147

File: Cone_spanners_2/test/Cone_spanners_2/CMakeLists.txt (hunk 1 above)
Comment (sloriot) [reply Cone_spanners_2/test/Cone_spanners_2/CMakeLists.txt:14]: We can open an independent issue if we want to fix that. OPTIONAL keyword should indeed be used.

**Pull request merged**